internal/bminventory/inventory.go

```diff
@@ -1031,9 +1031,17 @@ func (b *bareMetalInventory) InstallClusterInternal(ctx context.Context, params
 	if cluster, err = common.GetClusterFromDBWithHosts(b.db, params.ClusterID); err != nil {
 		return nil, common.NewApiError(http.StatusNotFound, err)
 	}
+
+	// Verify cluster is ready to install
+	if ok, reason := b.clusterApi.IsReadyForInstallation(cluster); !ok {
```

Comment:
Why was this added? Seems like this would prevent the cluster from installing in a case where it wasn't ready because roles needed to be assigned, but the code added below looks like it would handle exactly that situation.

Reply:
This should not be called if cluster is not ready.  When cluster is not ready we don't know if it can be fixed by calling auto-assign.
Anyway, roles should be set by the refresh-role. Here https://github.com/openshift/assisted-service/blob/424cea808c11a43eee184437bac03e851849f894/internal/host/monitor.go#L145 , so the roles in this stage should have already set.
We might be able to remove the code to auto-assign here, but I was a bit careful...

Reply:
In addition, auto assign role does not prevent the cluster to become ready. Here https://github.com/openshift/assisted-service/blob/424cea808c11a43eee184437bac03e851849f894/internal/cluster/validator.go#L419-L436